src/main/java/sirius/biz/storage/layer3/VirtualFile.java

```diff
@@ -247,6 +251,37 @@ public String relativePath(VirtualFile rootParent) {
         return result;
     }
 
+    /**
+     * Returns the root directory path of the current virtual file.
+     * <p>
+     * If the virtual file belongs to a known space, it is used as root, otherwise <tt>/</tt> is returned.
+     *
+     * @return the root directory path
+     */
+    @Nullable
+    public VirtualFile root() {
+        List<VirtualFile> fileList = pathList();
+        if (fileList.isEmpty()) {
+            return null;
+        }
+
+        if (fileList.size() > 1 && objectStorage.isKnown(fileList.get(1).name())) {
+            return fileList.get(1);
+        }
+
+        return fileList.get(0);
```

Comment:
I'd preface this block of code with a small inline comment, something like
`// index 0 is always '/' -  and known spaces will be at index 1`

just to clarify the structure of the list